### PR TITLE
Adapt ResamplingFunction interface to Python.Enum

### PR DESCRIPTION
**Pull request by phillip-wenig-frequenz**
solves #4 

The `ResamplingFunction` has new methods to adapt its interface to the python enum interface.

The following methods are added:

- values
- members
- __str__
- __repr__
- name
- value

### CODE REVIEW

File: frequenz/resampling/_rust_backend.pyi [hunk 1]
```diff
@@ -4,27 +4,44 @@
 __all__ = "Resampler", "ResamplingFunction"
 
 from datetime import datetime, timedelta
+from enum import Enum
 from typing import Optional
 
-class ResamplingFunction:
+class ResamplingFunction(Enum):
```
Comment (llucax) [inline frequenz/resampling/_rust_backend.pyi:11]:
```suggestion
@unique
class ResamplingFunction(Enum):
```
now? :D

File: src/python.rs [hunk 2]
```diff
@@ -41,6 +42,96 @@ enum ResamplingFunctionF32 {
...
+            (Self::Count.to_string(), Self::Count.value()),
+        ]
+    }
+
+    fn __str__(&self) -> String {
+        format!("{}.{}", "ResamplingFunction", self)
+    }
+
+    fn __repr__(&self) -> String {
+        self.__str__()
```
Comment (llucax) [inline src/python.rs]: The canonical `repr()` for `Enum` is:

```suggestion
        format!("<{}.{}: {}>", "ResamplingFunction", self.to_string(), self)
```

Maybe we can use that instead?

**Review by llucax — COMMENTED** [review]
Awesome that this works!

File: src/python.rs (hunk 2 above)
Comment (phillip-wenig-frequenz) [reply src/python.rs]: I would change it to:
```rust
fn __repr__(&self) -> String {
    format!("<{}: {}>", self.__str__(), self.value())
}
```

which would do this: 
```python
repr(ResamplingFunction.Average) == "<ResamplingFunction.Average: 0>"
```

Comment (llucax) [reply src/python.rs]: Yeah, I was about to suggest that but didn't know what's the most common approach in rust :)

**Review by llucax — APPROVED** [review]

**Pull request merged**